native/packages/react-native-bpk-component-button/src/utils.js

```diff
@@ -0,0 +1,146 @@
+  if (styles.types[type] && styles.types[type][elementType]) {
+    styleForElement.push(styles.types[type][elementType]);
+  }
+
+  // Add modifiers. Disabled comes last to override other styles.
+  if (large) {
+    let largeModifier = 'large';
+    if (['secondary', 'destructive'].includes(type)) {
+      largeModifier = 'largeWithOutline';
+    }
+    styleForElement.push(styles.modifiers[largeModifier][elementType]);
+  }
+
+  if (disabled) {
+    styleForElement.push(styles.modifiers.disabled[elementType]);
+  }
+
+  if (iconOnly) {
+    styleForElement.push(large ? styles.modifiers.iconOnlyLarge[elementType] : styles.modifiers.iconOnly[elementType]);
+  } else if (title && icon) {
+    // If it has a title and icon, get the style for that.
+    styleForElement.push(styles.modifiers[large ? 'textAndIconLarge' : 'textAndIcon'][elementType]);
+  }
+
+  return styleForElement;
+};
+
+export const getThemingForElement = (elementType, theme, { type, disabled }) => {
+  const themeForElement = {};
+  if (theme && !disabled && styles.themeMappings[elementType]) {
+    Object.keys(styles.themeMappings[elementType]).forEach((key) => {
+      const values = styles.themeMappings[elementType][key];
+      if (values[type]) {
+        themeForElement[key] = theme[values[type]];
+      }
+    });
+  }
+  return themeForElement;
+};
+
+export const getGradientColors = (theme, { type, disabled }) => {
+  let gradientColors = styles.gradientColors[type];
+  if (theme) {
+    const gradientThemeProps = styles.themeMappings.gradient[type];
+    gradientColors = [theme[gradientThemeProps.startColor], theme[gradientThemeProps.endColor]].filter(item => !!item);
+  }
+
+  if (disabled) {
+    gradientColors = styles.gradientColors.disabled;
+  }
+  return gradientColors;
+};
+
+export const iconPropType = (props, propName, componentName) => {
+  const icon = props[propName];
+  if (props.iconOnly && !icon) {
+    return new Error(`Invalid prop \`${propName}\` supplied to \`${componentName}\`. When \`iconOnly\` is enabled, \`${propName}\` must be supplied.`); // eslint-disable-line max-len
+  }
+  return false;
+};
+
+export const isTypeThemeable = type => THEMEABLE_TYPES.includes(type);
+
+export const themeAttributesSupplied = (type, theme) => (
+  difference(REQUIRED_THEME_ATTRIBUTES[type], Object.keys(theme)).length === 0
+);
+
+export const themePropType = (props, propName, componentName) => {
+  const type = props.type;
+  const theme = props.theme;
+  if (!theme) {
+    return false;
+  }
+  if (!themeAttributesSupplied(type, theme)) {
+    return new Error(`Invalid prop \`${propName}\` supplied to \`${componentName}\`. For buttons of type \`${type}\`, the \`theme\` prop must include \`${REQUIRED_THEME_ATTRIBUTES[type].join(', ')}\``); // eslint-disable-line max-len
+  }
+  return false;
+};
+
+export const getAndroidBackgroundColour = (theme, props) => {
+  const style = {
+    backgroundColor: 'transparent',
```

Comment:
This part of the style isn't cached since it's not created with `StyleeSheet` create right?

Reply:
Oh, nice catch! I'll amend this!